src/components/UserForm/validators.js

```diff
@@ -8,28 +8,23 @@ export const useDebouncedUniqueUsernameValidator = ({
     username: currentUsername,
 }) => {
     const engine = useDataEngine()
-    const findUserByUsername = pDebounce(async (username) => {
-        const {
-            users: { users },
-        } = await engine.query({
-            users: {
-                resource: 'users',
-                params: {
-                    filter: `userCredentials.username:eq:${username}`,
-                    fields: 'id',
-                },
+    const checkUsername = pDebounce(async (username) => {
+        const { result } = await engine.query({
+            result: {
+                resource: 'account/username',
+                params: { username },
             },
         })
-        return users[0]
+        return result
     }, 350)
     const validator = async (username) => {
         if (username === currentUsername) {
             return
         }
 
         try {
-            const user = await findUserByUsername(username)
-            if (user) {
+            const { response } = await checkUsername(username)
```

Comment:
I know this is not related to this PR but I find this API kind of weird. `message, response`... I would've thought `response` would've been a wrapper. Actually at first glance I thought there was an error in the code because I thought you wanted `result` not `response`.